cli/popper/scm.py

```diff
@@ -1,51 +1,40 @@
 import os
 import popper.utils as pu
+import git
 
 
-def get_scm_service_url():
-    return 'https://github.com/'
-
-
-def get_root_folder(debug=False):
-    """Tries to find the root folder,
-    """
-    return pu.exec_cmd('git rev-parse --show-toplevel', debug=debug)[0]
-
-
-def infer_repo_name_from_root_folder(debug=False):
+def infer_repo_name_from_root_folder(repo):
     """Finds the root folder of a local Github repository and returns it.
 
     Returns:
         repo_name (str): the name of the root folder.
     """
-    root_folder = get_root_folder(debug)
-    repo_name = os.path.basename(root_folder)
-    return repo_name
+    return os.path.basename(repo.git.rev_parse('--show-toplevel'))
 
 
-def get_name(debug):
+def get_name(repo):
     """Assuming $PWD is part of a Git repository, it tries to find the name of
     the repository by looking at the 'origin' git remote. If no 'origin' remote
     is defined, it returns the name of the parent folder.
     """
-    url = get_remote_url(debug)
+    url = get_remote_url(repo)
 
     if not url:
-        return infer_repo_name_from_root_folder()
+        return infer_repo_name_from_root_folder(repo)
 
     if url.endswith('.git'):
         url = url[:-4]
 
     return os.path.basename(url)
 
 
-def get_user(debug):
+def get_user(repo):
```

Comment:
can this be done via the GitPython library?

Reply:
@ivotron I couldn't find any direct API's for this